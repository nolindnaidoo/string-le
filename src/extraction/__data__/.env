# Application configuration
APP_NAME=My Production Application
APP_VERSION=2.1.3
APP_ENV=production
DEBUG=false
PORT=8080
TIMEOUT_SECONDS=30
MAX_RETRIES=5
RATE_LIMIT=1000
PI_VALUE=3.14159
RESPONSE_TIMEOUT=15.5

# Database settings  
DATABASE_URL=postgresql://admin:secret123@db.example.com:5432/myapp_prod
DB_HOST=db-cluster.internal
DB_PORT=5432
DB_NAME=production_database
DB_USERNAME=app_user
DB_PASSWORD="complex!@#$%password"
DB_SSL=true
DB_POOL_SIZE=50
DB_TIMEOUT=30

# API Configuration
API_BASE_URL=https://api.example.com/v2
API_KEY="Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"
API_TIMEOUT=5000
API_RETRIES=3
WEBHOOK_SECRET="whsec_1234567890abcdef"
OAUTH_CLIENT_ID=oauth_12345
OAUTH_CLIENT_SECRET="oauth_secret_abcdef"

# JSON-like configuration strings
FEATURE_FLAGS='{"analytics": true, "chat": false, "notifications": true}'
USER_PERMISSIONS='["read", "write", "admin"]'
CORS_ORIGINS='["https://app.example.com", "https://admin.example.com"]'

# URLs and endpoints
REDIS_URL=redis://localhost:6379/0
ELASTICSEARCH_URL=https://search.example.com:9200
WEBHOOK_URL=https://hooks.example.com/webhooks/abc123
CDN_BASE_URL=https://cdn.example.com/assets

# Feature toggles and booleans
ENABLE_ANALYTICS=true
ENABLE_LOGGING=false
MAINTENANCE_MODE=false
USE_CACHE=true
SEND_EMAILS=true
BETA_FEATURES=false

# Numeric values (should be extracted as strings)
MAX_USERS=50000
MIN_PASSWORD_LENGTH=8
SESSION_TIMEOUT=3600
FILE_SIZE_LIMIT=104857600
CACHE_TTL=1800

# Complex strings with various content
WELCOME_MESSAGE="Welcome to our enterprise platform!"
ERROR_MESSAGE='Something went "terribly wrong" during processing'
SUCCESS_MESSAGE="Operation completed successfully ‚úì"
NOTIFICATION_TEXT="You have 5 new messages"
SYSTEM_ALERT='CRITICAL: Database connection failed!'

# Special characters and encoding
SPECIAL_CHARS="Special: !@#$%^&*()_+-=[]{}|;:,.<>?"
UNICODE_TEXT="Hello ‰∏ñÁïå! –ó–¥—Ä–∞–≤—Å—Ç–≤—É–π –º–∏—Ä! üåç"
EMAIL_TEMPLATE="Dear {{name}}, welcome to {{app_name}}!"
SQL_QUERY="SELECT * FROM users WHERE active = true AND role IN ('admin', 'user')"

# Edge cases and problematic values
EMPTY_VALUE=
NULL_VALUE=null
ZERO_VALUE=0
FALSE_VALUE=false
TRUE_VALUE=true
QUOTED_EMPTY=""
QUOTED_SPACES="   "
MIXED_QUOTES="She said 'Hello \"World\"!' to everyone"

# Environment-specific overrides
NODE_ENV=production
PYTHON_PATH=/usr/local/bin/python3
LOG_LEVEL=info
LOG_FORMAT=json
TZ=UTC
LANG=en_US.UTF-8

# Comments and invalid lines (should be ignored)
# This is a comment and should be ignored
# COMMENTED_VAR=this should not appear
INVALID_LINE_NO_EQUALS_SIGN
= EMPTY_KEY_NOT_VALID
TRAILING_SPACES=value with trailing spaces   

# Export syntax variations
export EXPORTED_VAR="exported variable value"
export EXPORTED_NUMBER=42
export EXPORTED_BOOL=true
