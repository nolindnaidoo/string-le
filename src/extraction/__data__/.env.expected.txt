["https://app.example.com", "https://admin.example.com"]
["read", "write", "admin"]
{"analytics": true, "chat": false, "notifications": true}
/usr/local/bin/python3
0
1000
104857600
15.5
1800
2.1.3
3
3.14159
30
3600
42
5
50
5000
50000
5432
8
8080
app_user
Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9
complex!@#$%password
CRITICAL: Database connection failed!
db-cluster.internal
Dear {{name}}, welcome to {{app_name}}!
EMPTY_KEY_NOT_VALID
en_US.UTF-8
exported variable value
false
Hello ‰∏ñÁïå! –ó–¥—Ä–∞–≤—Å—Ç–≤—É–π –º–∏—Ä! üåç
https://api.example.com/v2
https://cdn.example.com/assets
https://hooks.example.com/webhooks/abc123
https://search.example.com:9200
info
json
My Production Application
null
oauth_12345
oauth_secret_abcdef
Operation completed successfully ‚úì
postgresql://admin:secret123@db.example.com:5432/myapp_prod
production
production_database
redis://localhost:6379/0
SELECT * FROM users WHERE active = true AND role IN ('admin', 'user')
She said 'Hello \"World\"!' to everyone
Something went "terribly wrong" during processing
Special: !@#$%^&*()_+-=[]{}|;:,.<>?
true
UTC
value with trailing spaces
Welcome to our enterprise platform!
whsec_1234567890abcdef
You have 5 new messages
